report_qweb_pdf_watermark/models/report.py

```diff
@@ -108,11 +111,29 @@ def _run_wkhtmltopdf(
             return result
 
         pdf = PdfFileWriter()
-        pdf_watermark = None
+
         try:
             pdf_watermark = PdfFileReader(BytesIO(watermark))
         except PdfReadError:
-            # let's see if we can convert this with pillow
+            pdf_watermark = None
+        except UnicodeDecodeError as e:
+            # workaround: handle case of unintentional pypdf `UnicodeDecodeError`
+            # because odoo enforces `strict` mode
+            # https://github.com/versada/odoo/blob/d83bd4f/odoo/tools/pdf.py#L27
+            # and `pypdf` validation may can raise the error when generating message
+            # https://github.com/py-pdf/pypdf/blob/2.x/PyPDF2/_reader.py#L1455
+            stream = BytesIO(watermark)
+            stream.seek(0, os.SEEK_SET)
+            header_byte = stream.read(5)
+            if header_byte != b"%PDF-":
+                # the file is not PDF -- probably this is image (we'll check bellow)
```

Comment:
Is this module supposed to handle images as well? Cause from name I would assume it only handles pdfs:)

Reply:
yes, 
- [code](https://github.com/versada/reporting-engine/pull/7/files#diff-66dceb063022cd58404dbbbb4649ffec677cbc89be10c385ad5714231929a220R138-R147)
- [test](https://github.com/versada/reporting-engine/blob/17.0-mig-report_qweb_pdf_watermark/report_qweb_pdf_watermark/tests/test_report_qweb_pdf_watermark.py#L22)